src/wazuh_modules/wm_aws.c

```diff
@@ -39,40 +39,28 @@ const wm_context WM_AWS_CONTEXT = {
 void* wm_aws_main(wm_aws *aws_config) {
     wm_aws_bucket *cur_bucket;
     wm_aws_service *cur_service;
-    time_t time_start;
-    time_t time_sleep = 0;
     char *log_info;
 
 
     wm_aws_setup(aws_config);
     mtinfo(WM_AWS_LOGTAG, "Module AWS started");
 
-    // First sleeping
+    // Main loop
 
-    if (!aws_config->run_on_start) {
-        time_start = time(NULL);
+    do {
 
-        // On first run, take into account the interval of time specified
-        if (aws_config->state.next_time == 0) {
-            aws_config->state.next_time = time_start + aws_config->interval;
-        }
+        const time_t time_sleep = sched_scan_get_next_time(&(aws_config->scan_config), WM_AWS_LOGTAG, aws_config->run_on_start);
+
+        aws_config->state.next_time = aws_config->scan_config.time_start + time_sleep;
```

Comment:
`aws_config->state.next_time` is used as a mechanism to ensure that a restart of the module does not affect the remaining time to be launched (when run on start is disabled). Its value is read with `wm_state_io()` at startup, so, it should not be overwritten in case it is not NULL.
Please change this logic for all the affected modules.

Reply:
Alright, I have changed it now so it only overrides it when `state.next_time == 0`